keps/349-avoid-coscheduling/README.md

```diff
@@ -0,0 +1,350 @@
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+Some workloads need all pods to be running at the same time to make progress, for
+example, when they require pod-to-pod communication. In that case a large
+workload may have difficulty scheduling all of its pods for a prolonged time as
+smaller workloads would continue to block the slots. Moreover, a pair of
+workloads may deadlock if the amount of actual resources available is below the
+quota configured to determine admission.
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+- a mechanism to avoid concurrent scheduling of pods for workloads.
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+- guarantee that two workloads would not schedule pods concurrently. Example
+scenarios in which two workloads may still concurrently schedule their pods:
+  - when succeeded pods are replaced with new because job's parallelism is less than its completions;
+  - when a failed pod gets replaced.
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+We introduce a mechanism to mark a Job as blocking admission of further
+workloads until the first batch of its pods is scheduled, more precisely, until:
+`job.status.ready + job.status.succeeded >= job.spec.parallelism`.
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a machine learning researcher I want to run a large workload, comprising
+thousands of pods which need to all run at the same time. Scheduling all pods
+in this workload takes up to 20min and I want to have a guarantee that other
```

Comment:
20 min doesn't sound realistic.

The more realistic scenario is with long running jobs with tens of pods. Autoscaling+scheduling could take a couple of minutes. The thing is that autoscaling could face stockouts.

Reply:
Adjusted the description